benchmarks/webdataset_hub_benchmarks.ipynb

```diff
@@ -0,0 +1,342 @@
+        "            # Read the JPEG-compressed image file contents.\n",
+        "            image = readfile(fname)\n",
+        "\n",
+        "            # Construct a uniqu keye from the filename.\n",
+        "            key = os.path.splitext(os.path.basename(fname))[0]\n",
+        "\n",
+        "            # Useful check.\n",
+        "            assert key not in all_keys\n",
+        "            all_keys.add(key)\n",
+        "\n",
+        "            # Construct a sample.\n",
+        "            xkey = key if filekey else \"%07d\" % i\n",
+        "            sample = {\"__key__\": xkey, \"jpg\": image, \"cls\": cls}\n",
+        "\n",
+        "            # Write the sample to the sharded tar archives.\n",
+        "            sink.write(sample)\n",
+        "\n",
+        "for split in splits:\n",
+        "    print(\"# split\", split)\n",
+        "    write_dataset(data, base=shards, split=split)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "bWdSmBcLSiWP"
+      },
+      "source": [
+        "# Timing the Read Access of WebDataset\n",
+        "\n",
+        "Now that the dataset has been sharded for WebDataset, we can start making the dataloaders to iterate over the dataset and time the read access overhead."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "WYkTeHXFHw4a"
+      },
+      "source": [
+        "def time_webdataset(url, batch_size=64):\n",
+        "    dataset = wds.Dataset(url)\n",
+        "    loader = torch.utils.data.DataLoader(dataset, batch_size=batch_size, num_workers=8)\n",
+        "    start = time.time()\n",
+        "    for filename, inputs, targets in loader:\n",
+        "        x, y = inputs, targets\n",
+        "    end = time.time()\n",
+        "    print(\"Time taken by WebDataset: \", end - start)"
+      ],
+      "execution_count": null,
+      "outputs": []
+    },
+    {
+      "cell_type": "markdown",
+      "metadata": {
+        "id": "PfUvvCsKS0Uv"
+      },
+      "source": [
+        "# Timing the Read Access of Hub\n",
+        "\n",
+        "The following cell aims to time the package \"Hub\" for the same read access overhead."
+      ]
+    },
+    {
+      "cell_type": "code",
+      "metadata": {
+        "id": "RS8LLMKLSzyf"
+      },
+      "source": [
+        "def time_hub(tag, batch_size=64):\n",
+        "    dataset = hub.Dataset(tag)\n",
+        "    start = time.time()\n",
+        "    for batch in range(dataset.shape[0] // batch_size):\n",
+        "        x = dataset[\"image\"][batch * batch_size : (batch + 1) * batch_size].compute()\n",
+        "        y = dataset[\"label\"][batch * batch_size : (batch + 1) * batch_size].compute()\n",
+        "    if dataset.shape[0] % batch_size != 0:\n",
+        "        x = dataset[\"image\"][(dataset.shape[0] // batch_size) * batch_size : ].compute()\n",
```

Comment:
we should make the comparison as similar to the original as possible. only batches are compared. there's no need to add a timing over a iteration over the entire dataset

Reply:
So, should I just write a pass command within the for loops?

Reply:
I decided to remove this function and I'll likely merge it in this state. When it gets merged, feel free to create a new pull request adding this function if it is deemed as needed. The only caveat is that it needs to be parametrized so that we can run it on varying number of workers.